api/resources_portal/test/end_to_end/test_multiple_resources_requested_and_fulfilled.py

```diff
@@ -0,0 +1,262 @@
+        OrganizationUserSetting.objects.create(user=self.post_doc, organization=self.primary_lab)
+
+        Notification.objects.all().delete()
+
+    @patch("orcid.PublicAPI", side_effect=generate_mock_orcid_record_response)
+    @patch("requests.post", side_effect=generate_mock_orcid_authorization_response)
+    def test_create_account_and_list_resource(self, mock_auth_request, mock_record_request):
+        # Create account (Requester)
+        self.client.get(get_mock_oauth_url([]))
+        requester = User.objects.get(pk=self.client.session["_auth_user_id"])
+
+        # Search resources
+        response = self.client.get(reverse("search-materials-list"), {"organization": "PrimaryLab"})
+
+        search_json = response.json()["results"]
+        chosen_materials_json = search_json[0:2]
+
+        # Select two resources from PrimaryLab and request them, uploading two signed IRBs
+        self.client.force_authenticate(user=requester)
+
+        material1 = Material.objects.get(pk=chosen_materials_json[0]["id"])
+        material2 = Material.objects.get(pk=chosen_materials_json[1]["id"])
+
+        request1 = MaterialRequest(material=material1, requester=requester)
+        request2 = MaterialRequest(material=material2, requester=requester)
+
+        response = self.client.post(
+            reverse("material-request-list"), model_to_dict(request1), format="json"
+        )
+        request_1_id = response.data["id"]
+        response = self.client.post(
+            reverse("material-request-list"), model_to_dict(request2), format="json"
+        )
+        request_2_id = response.data["id"]
+
+        self.assertEqual(
+            len(
+                Notification.objects.filter(
+                    notification_type="TRANSFER_REQUESTED", email=self.post_doc.email
+                )
+            ),
+            2,
+        )
+
+        # Upload IRBs
+        irb1 = Attachment(
+            filename="executed_mta",
+            description="Executed transfer agreement for the material.",
+            s3_bucket="a bucket",
+            s3_key="a key",
+        )
+
+        irb2 = Attachment(
+            filename="executed_mta",
+            description="Executed transfer agreement for the material.",
+            s3_bucket="a bucket",
+            s3_key="a key",
+        )
+
+        response = self.client.post(reverse("attachment-list"), model_to_dict(irb1), format="json")
+        irb_1_id = response.data["id"]
+        response = self.client.post(reverse("attachment-list"), model_to_dict(irb2), format="json")
+        irb_2_id = response.data["id"]
+
+        self.client.put(
+            reverse("material-request-detail", args=[request_1_id]), {"irb_attachment": irb_1_id}
+        )
+        self.client.put(
+            reverse("material-request-detail", args=[request_2_id]), {"irb_attachment": irb_2_id}
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```

Comment:
@davidsmejia I think I recall you saying the plan was to upload materials first, then create the request? Is that correct?